packages/plugin-core/src/windowWatcher.ts

```diff
@@ -122,6 +125,17 @@ export class WindowWatcher {
     return;
   }
 
+  /**
+   * Show note preview panel if applicable
+   */
+  async triggerNotePreviewUpdate({ document }: TextEditor) {
+    const maybeNote = WSUtils.tryGetNoteFromDocument(document);
+    if (maybeNote)
+      PreviewPanelFactory.getProxy().showPreviewAndUpdate(maybeNote);
```

Comment:
Rather than calling the static `getProxy()` (which was meant only as a temporary method until we could finish refactoring), let's try to pass a instance of `PreviewProxy` into the constructor of `WindowWatcher`.  Then, in `workspace.ts`, we can pass in the static instance when constructing `WindowWatcher` (for now).

Then in testing, we can just use a dummy implementation of `PreviewProxy`.

However, if we want to do this show on window switching  functionality, we will still need to expose a `show()` method on `PreviewProxy`

Reply:
Good idea, I'll update the constructor